app/pages/index.tsx

```diff
@@ -7,12 +7,28 @@ import DefaultLayout from 'layouts/default-layout';
 import RegistrationLoginButtons from 'containers/RegistrationLoginButtons';
 import {GUEST} from 'data/group-constants';
 import KeyDates from 'containers/KeyDates';
+import {getUserGroups} from 'server/helpers/userGroupAuthentication';
+import {getUserGroupLandingRoute} from 'lib/user-groups';
 
 const ALLOWED_GROUPS = [GUEST];
 
 interface Props extends CiipPageComponentProps {
   query: pagesQueryResponse['query'];
 }
+
+export async function getServerSideProps(context) {
+  const groups = getUserGroups(context.req);
+  const landingRoute = getUserGroupLandingRoute(groups);
+  if (landingRoute === '/')
+    return {
+      props: {}
+    };
+
+  return {
+    redirect: {destination: landingRoute, permanent: false}
+  };
+}
+
```

Comment:
I'm pretty sure the routing needs to be done in `server/index.js` - even using getServerSideProps, this only runs when you are on the homepage.

So if I'm on an application page for example and click "Dashboard", I don't hit this until I first load the homepage on the client.

Reply:
(in other words, using getServerSideProps only makes the homepage load faster, and thus perform the redirect a bit faster)

Reply:
`getServerSideProps` is run by the next middleware on the server side, so this should work

Reply:
Note to self: This solution doesn't work in development when using the `AS_REPORTER` etc. flags, which won't be the case on production.